### ISSUE TITLE
Typecover does not work on Surface Pro 3

### ISSUE THREAD

**Issue by dplrs** [linux-surface/linux-surface#311]
This is not exactly Linux related issue, but I believe I can get much more info here than anywhere else. My Type Cover does not work on my Surface Pro 3 (neither on the Windows, nor in Linux and not even in UEFI boot menu) , but it is functional on other Surface Pro machines. The issue started after some Windows Update. The Type Cover stopped working all of a sudden. None of the troubleshooting methods suggested by Microsoft on their website solved the problem. Then I came across a Microsoft Community Forum Thread where a possible theory for what might be the problem was: 

> There are unfortunately many people with this issue (including a neighbor and a friend). For most, the problem occurs after some random Windows Update. My theory is that it isn't a particular update, but something in the update Process that corrupts the firmware somehow. Microsoft has not commented on this, nor have they come up with a fix. If it does not work in the UEFI, no amount of reimaging will fix it. Cleaning the contacts has not fixed this for anyone.

I'm pretty sure this is a software problem, since there are so many people having the same issue online. Since linux gives much more fine-grained control than Windows, I was wondering if there are any ways I can go around looking for where the problem lies. I looked at the dmesg output whenever I attached or detached the Type cover, but nothing is printed.

### Environment

- Hardware model: Surface Pro 3
- Kernel version: 5.8.15-surface
- Distribution: Ubuntu 20.04

**Comment by qzed** [comment]
Based on your description, you're probably correct that it's some kind of firmware problem. You could maybe dump and decompile your ACPI tables and compare the functions that fail above that to the code we have in https://github.com/linux-surface/acpidumps/tree/master/surface_pro_3 (which is probably from an older firmware revision).

You could also try to ask some other Pro 3 users (see e.g. https://github.com/linux-surface/linux-surface/wiki/Testers-and-Contributing) if they've installed the latest firmware update and if not get them to run the fwts to compare.

I'm not sure about the Pro 3, but on newer devices it's also possible to re-install the firmware with fwupd (https://github.com/linux-surface/surface-uefi-firmware). No guarantees on that though. Worst-case, this might brick your device if something goes wrong.

**Comment by Cyborgscode** [comment]
by the quality of that bios .. bricking is not so unlikely ;) 

BTW: A pro3 should work completely with a Fedora mainstream kernel, at least the device i had to test, did.

**Comment by dplrs** [comment]
@Cyborgscode But the keypad doesn't work in UEFI menu itself. So, it doesn't matter which kernel or distro I run, right?

**Comment by dplrs** [comment]
@benajminfaul2 @Newbytee @RussH Hi, Can you please run the Firmware Test Suite Batch test on your Surface Pro 3 devices with Type Cover attached using the `sudo fwts --batch` command, and post the results.log, so I can compare the results. It'd be a great help. Thanks

**Comment by Newbytee** [comment]
> @benajminfaul2 @Newbytee @RussH Hi, Can you please run the Firmware Test Suite Batch test on your Surface Pro 3 devices with Type Cover attached using the `sudo fwts --batch` command, and post the results.log, so I can compare the results. It'd be a great help. Thanks

Here you go: [results.log](https://github.com/linux-surface/linux-surface/files/5420920/results.log)

**Comment by dplrs** [comment]
> > @benajminfaul2 @Newbytee @RussH Hi, Can you please run the Firmware Test Suite Batch test on your Surface Pro 3 devices with Type Cover attached using the `sudo fwts --batch` command, and post the results.log, so I can compare the results. It'd be a great help. Thanks
> 
> Here you go: [results.log](https://github.com/linux-surface/linux-surface/files/5420920/results.log)

Thanks a lot, if it's not much trouble. Can you also post the output of `sudo fwts --utils`?

**Comment by Newbytee** [comment]
> @Newbytee Thanks a lot, if it's not much trouble. Can you also post the output of `sudo fwts --utils`?

Coming right at you: [results.log](https://github.com/linux-surface/linux-surface/files/5421644/results.log)

**Comment by RussH** [comment]
and just so you have another set to compare with;
[fwts-batch-results.log](https://github.com/linux-surface/linux-surface/files/5421987/fwts-batch-results.log)
[fwts-utils-result.log](https://github.com/linux-surface/linux-surface/files/5421988/fwts-utils-result.log)

If the hardware shows different on mine - it's because the internal wifi/bluetooth died, and I'm now using a usb wifi adaptor

**Comment by Cyborgscode** [comment]
are you sure, your hw isn't breaking apart?

> @Cyborgscode But the keypad doesn't work in UEFI menu itself. So, it doesn't matter which kernel or distro I run, right?

Yes. Last time this happend to me. my keypad was dead. A new one fixed the issue, but as you said, you tested your on a different Surface and it worked. Did you recheck this with the other typcover on your device? 

It's also possible, that the connector chip died. This hw is fragile ;) my Surface once stopped it's fans for 24hs and now, they work as ever. Judgeing by your report, that wifi is gone too, a hw malfunction is a possability.

**Comment by dplrs** [comment]
> Did you recheck this with the other typcover on your device?

Yes, I took another functional typecover and plugged it into my Surface Pro, it didn't work. But everthing else is works fine. I'm not sure if it's a hardware failure, since many people online have reported the same0 problem where their surface pro suddenly fails to to detect the attached type cover, This has to do something with some windows update which corrupts the firmware somehow.

**Comment by algeorgiadis** [comment]
I have the exact same problem in fedora 38 on a surface pro 4. Did you find out what the problem was finally @dplrs? I have tested with 2 type covers, and none of it seems to work (no lights - no keys - no touch pad) even in the uefi menu.

**Comment by Cyborgscode** [comment]
> I have the exact same problem in fedora 38 on a surface pro 4. Did you find out what the problem was finally @dplrs? I have tested with 2 type covers, and none of it seems to work (no lights - no keys - no touch pad) even in the uefi menu.

most likely a hardware defect. As a last resort test.. let the surface ran out of energy, and i mean really ran out to 0.000000. keep it in this state for an hour and then recharge the battery and boot into the bios setup menue. IF anything in the volatile memory is causing the malfunction of the hardware, this -> could <- reset it back to default. If it does not change a thing, the possibility that your hw is defect is very high.  

You all will find, that spare parts for surface device are cheap compared to the whole device, the only bad about replacing one is, you need to lift the touchscreen with a "thermal border", which melts the glue and opens the device without breaking or damaging the touchscreen. The replacement it self, is a piece of cake. you can do it in around 5 minutes.

**Comment by algeorgiadis** [comment]
Thank you @Cyborgscode. I will let it drain, and I will try again, because it happened, after a complete drain of the battery. Let's hope for the easy fix!!
